services/data/src/engine/types/Query.ts

```diff
@@ -19,7 +18,11 @@ export interface ResolvedResourceQuery extends ResourceQuery {
 }
 
 export type Query = Record<string, ResourceQuery>
-export type QueryResult = JsonMap
+export type QueryResult = any
```

Comment:
We could potentially use `unknown` here instead, but then we are enforcing consumers to type their data. So you couldn`t do `dataElements.dataElements.map()`. So if we were going this route, we might as well enforce the consumer to pass in the `TQueryResultData` type to `useDataQuery`. 
If we used `unknown` you would have to do something like this

```typescript
![image](https://user-images.githubusercontent.com/13852438/236454975-686833d8-0ec5-4339-ad14-9fa1bef74077.png)


```